app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -161,8 +161,13 @@ class PendingTransactionTracker extends EventEmitter {
 
       return
     }
-    // *note to self* hard failure point
-    const transactionReceipt = await this.query.getTransactionReceipt(txHash)
+
+    let transactionReceipt
+    try {
+      transactionReceipt = await this.query.getTransactionReceipt(txHash)
+    } catch (_) {
+      transactionReceipt = undefined
```

Comment:
Also the error is now being swallowed, leaving us with no idea what went wrong.

Prior to #7483 the error message was at least logged somewhere (in `txMeta.history`, because it was assigned to the `warning` property). And after that it was thrown. Now it's recorded nowhere. This seems like a step backward.